test/e2e/collect-logs.sh

```diff
@@ -34,41 +34,124 @@ echo -e "${GREEN}Starting log collection at ${TIMESTAMP}${NC}"
         return
     fi
-    
-    # Collect logs for each container
-    for container in $containers; do
-        log_file="${log_file_prefix}-${container}.log"
-        echo "  - Container ${container} -> ${log_file}"
-        
-        # Get current logs
-        kubectl logs "${pod_name}" -n "${NAMESPACE}" -c "${container}" > "${log_file}" 2>&1 || \
-            echo "Failed to get logs for container ${container}" > "${log_file}"
-        
-        # Try to get previous logs if pod was restarted
-        previous_log_file="${log_file_prefix}-${container}-previous.log"
-        if kubectl logs "${pod_name}" -n "${NAMESPACE}" -c "${container}" --previous > "${previous_log_file}" 2>&1; then
-            echo "  - Previous logs for ${container} -> ${previous_log_file}"
-        else
-            rm -f "${previous_log_file}"
-        fi
+
+    # Create node logs directory
+    mkdir -p "${node_log_dir}"
+
+    for node in $nodes; do
+        echo "  - Collecting ${agent_type} logs from node ${node}"
+        local node_specific_dir="${node_log_dir}/${node}"
+        mkdir -p "${node_specific_dir}"
+
+        # Collect specific agent logs from node filesystem
+        collect_agent_logs_from_node "${node}" "${cluster_name}" "${node_specific_dir}" "${agent_type}"
     done
 }
 
+# Function to collect specific agent logs from node filesystem
+# Collects all log files including rotated (*.log.*) and compressed (*.gz) files
+# Args:
+#   node: The node name to collect logs from
+#   cluster_name: The cluster name for logging context
+#   node_log_dir: The directory to save the collected logs
+#   agent_type: The type of agent ("hub-agent" or "member-agent")
+collect_agent_logs_from_node() {
+    local node=$1
+    local cluster_name=$2
+    local node_log_dir=$3
+    local agent_type=$4  # "hub-agent" or "member-agent"
+
+    echo "    -> Collecting ${agent_type} logs from node filesystem"
+    echo "    -> Found log paths: $(docker exec "${node}" find /var/log/pods -path "*/fleet-system_*${agent_type}*")"
+
+    # First check if any agent logs exist on this node (including .log, .log.*, and .gz files)
+    local log_files
+    log_files=$(docker exec "${node}" find /var/log/pods -path "*/fleet-system_*${agent_type}*" -type f \( -name "*.log" -o -name "*.log.*" -o -name "*.gz" \) 2>/dev/null || echo "")
+
+    if [ -n "$log_files" ]; then
+
+        # Process each log file separately
+        echo "$log_files" | while read -r logfile; do
+            if [ -n "$logfile" ]; then
+
+                # Extract a meaningful filename from the log path
+                local base_path=$(basename "$(dirname "$logfile")")
+                local original_filename=$(basename "$logfile")
+                local sanitized_filename="${base_path}_${original_filename}"
+
+                # Remove .gz extension for the output filename if present
+                local output_filename="${sanitized_filename%.gz}"
+                # Ensure output filename ends with .log
```

Comment:
If there are only 2 cases, i.e. .gz and .log, maybe just handle the .gz case explicitly rather than appending .log for all cases where it doesn't end with .log? In case the file suffix is unexpected, we should log a warning because it might not even be a text file

Reply:
it could be  .log, .log.*, and .gz files.

added the debug info in line 85 just in case there will be other formats i'm not aware of (but low possibility).

line 89 we only copy these three kinds of files.